src/ai/ai_controller.c

```diff
@@ -21,14 +21,18 @@
 
 #include "ai_controller.h"
 
+#include <string.h>
+
 #include "main/rom.h"
 #include "memory/memory.h"
+#include "r4300/cp0.h"
+#include "r4300/interupt.h"
+#include "r4300/mi_controller.h"
```

Comment:
ok, sounds fair. I can modify it later (most likely not in the next few days due to some other tasks). Just don't merge it until I've updated the source.